_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -945,6 +945,50 @@ mfxStatus VideoDECODEVP9_HW::PrepareInternalSurface(UMC::FrameMemID &mid, mfxFra
     return MFX_ERR_NONE;
 }
 
+
+static mfxStatus CheckFrameInfo(mfxFrameInfo &currInfo, mfxFrameInfo &info)
+{
+    mfxStatus sts = CheckFrameInfoCommon(&info, MFX_CODEC_VP9);
+    MFX_CHECK(sts == MFX_ERR_NONE, sts);
+
+    switch (info.FourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_AYUV:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y410:
+#endif
+            break;
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+#endif
+           if (info.Shift != 1)
+              return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
+           break;
+        default:
+            return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
+    }
+
+    switch(info.ChromaFormat)
+    {
+        case MFX_CHROMAFORMAT_YUV420:
+        case MFX_CHROMAFORMAT_YUV400:
+        case MFX_CHROMAFORMAT_YUV422:
+        case MFX_CHROMAFORMAT_YUV444:
+            break;
+        default:
+            return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
```

Comment:
```
MFX_CHECK_STS(MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
```

Reply:
Do u really support 400/422?

Reply:
Why? sts == MFX_ERR_NONE here ... We don't need to check it.

Reply:
@IlyaSergeyev didn't get your idea